spec/models/mixins/archived_mixin_spec.rb

```diff
@@ -0,0 +1,90 @@
+RSpec.describe ArchivedMixin do
+  let(:model_name) { :container }
+  let(:klass) { Container }
+
+  let(:timestamp) { Time.now.utc }
+  let(:active_model) { FactoryBot.create(model_name) }
+  let(:archived_model) { FactoryBot.create(model_name, :deleted_on => timestamp) }
+
+  describe ".archived" do
+    it "fetches archived records" do
+      active_model
+      archived_model
+
+      expect(klass.archived).to eq([archived_model])
+    end
+  end
+
+  describe ".active" do
+    it "fetches active records" do
+      active_model
+      archived_model
+
+      expect(klass.active).to eq([active_model])
+    end
+  end
+
+  describe ".active_for" do
+    it "detects active are in the range" do
+      active_model
+      expect(klass.active_for(2.years.ago)).to eq([active_model])
+    end
+
+    it "detects not yet deleted records are in the range" do
+      archived_model
+      expect(klass.active_for(2.years.ago)).to eq([archived_model])
+    end
+
+    it "detects already deleted records are not in the range" do
```

Comment:
These `its` here make it obvious to me that the naming is tough.  I think this becomes easier to "describe" if it's `archived_before` or similar as it's clear, we're only looking for objects that have been deleted and the deletion date is before some value.

Reply:
so instead of active, we would use `not_archived_before`?

Reply:
As I mentioned in the [other thread](https://github.com/ManageIQ/manageiq/pull/22589#discussion_r1244400750), let's discuss over voice.  Switching to `archived_before` ignores the fact that created_on will play a part in the detection as well, it just happens to not play a part of the method right now.  That is, objects created _after_ will also be removed from the scope and that doesn't make sense with a method named `archived_before`.  That being said, maybe it does make sense to make the naming match the current semantics, and then change it later?  I'm not sure, hence why I want voice.

FWIW, I think the confusion lies in the word "deleted", because there are actually deleted record, and then there are archived records with a deleted_on value.

Reply:
I understand that we spoke about taking into account `created_on`. But that was when we were creating a VimPerformanceState in the past. If we create a record in the past, then new records will be created and we will need to make sure they do not get into the `state_data`.

Since we have agreed that for this current code, we will only run for `Time.now`, and we will never see future records. I wasn't able to come up with a use case where we would need to take into account `created_on`. So I felt ignoring that for now made sense.

If the logic to take into account were as simple as `created_on`, then that may make sense, but since we will not need that feature in the near future, adding that complexity did not make sense to me.

Reply:
ooh but yea, I liked the original name too.
`active` means not `archived` and therefore `deleted_on` is not set. But really, I guess that means `active(Time.now)`.

`active_on(date)` / `active_for(daterange)` means that on a particular date, it hasn't been deleted (and as discussed, it has been created / ...)

ASIDE: Another reason why I got out of the timerange game is because [`get_time_interval`](https://github.com/ManageIQ/manageiq/blob/master/app/models/metric/helper.rb#L180-L187) seems to suggest that the timeframe ends on the `VimPerformanceState#timestamp` rather than starts there. that confused me. Since this is more strategic, punting seemed like a plan.

Reply:
Is this resolved?